controlpanel/frontend/views/parameter.py

```diff
@@ -0,0 +1,63 @@
+from django.contrib import messages
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.urls import reverse_lazy
+from django.views.generic import ListView, CreateView
+from django.views.generic.edit import FormMixin, DeleteView
+
+from controlpanel.api.models import Parameter
+from controlpanel.api.permissions import is_superuser
+from controlpanel.frontend.forms import CreateParameterForm
+
+
+class ParameterList(LoginRequiredMixin, ListView):
+    model = Parameter
+    template_name = "parameter-list.html"
+
+    def get_queryset(self):
+        queryset = Parameter.objects.all()
+        if is_superuser(self.request.user):
```

Comment:
Hang on a second, isn't this logic already in the filter? Shouldn't be enough to tell this view to use that filter and the queryset would already be filter according to that? This duplication can lead to confusion/bugs.

Reply:
The filters you are referring to are for the REST API. I agree that the duplication should be avoided, but I think you'll have to do it via a model manager.

Reply:
I'm so confused. What's the difference between the views in the API and the views in the frontend? I know it must sound like a very stupid question...

Reply:
Right, so the views in `api` are gone and moved to `frontend`. Fair enough...but what happened to the filtering @andyhd?

[They used to be used](https://github.com/ministryofjustice/analytics-platform-control-panel/blob/85273c703da7caa0ed348aa1d4c8b2c96d8fef37/control_panel_api/views.py#L239)...right, looks like some of the views are now in `controlpanel/api/views/models.py`...that's why I couldn't find them.

Sorry, got confused by the new structure.

Reply:
Mmmm 🤔 but if this is reading from the DB, can't it just say "use the filter" and django would filter the records automatically according to the filter class? I haven't touched this in a long time so maybe I'm missing something obvious here.

Reply:
The old API views are still in place, just separated out into smaller files. They handle REST API URLs like `/api/cpanel/v1/apps/`, the new frontend views are for the web interface at URLs like `/webapps/`. It should be possible to share the filtering code between them, though.

Reply:
The filters you are talking about are Django REST Framework filters and are not designed to be used outside of the framework. But the `filter_queryset` method that they have is just Django ORM code, so it should be possible to share that.

Reply:
The DRF provides its own `View` classes which use the filters. But this frontend view is a different type of `View` class and doesn't have all the setup to use those filters. But, as I said, it shouldn't be too difficult to factor out the common code and share it.

Reply:
@andyhd I see what you mean.

That's interesting...aren't the views in the `frontend` getting the data using the models/DB directly tho? If that's the case, is the `api` even used? And if not, are the permissions even used anymore? Does it mean that the user can just URL browse and see/do whatever they want? 😱😱

For example, this is the [view to delete a user](https://github.com/ministryofjustice/analytics-platform-control-panel/blob/master/controlpanel/frontend/views/user.py#L43) it says to use the `User` model but where are [the permission](https://github.com/ministryofjustice/analytics-platform-control-panel/blob/master/controlpanel/api/permissions.py#L61) enforced 🙀?

Reply:
That is an excellent point. I'll fix the security in the existing views now.

Reply:
This restricts users from doing any action on parameters they haven't created - including delete - as the queryset only contains their own parameters. Unless you are a superuser of course.